1.0.0 RC1 blog post (#1214)
cake-build/website: input/blog/2020-11-05-cake-v1.0.0-rc0001-released.md

﻿---
title: Cake v1.0.0 RC1 released
category: Release Notes
author: devlead
---

Version 1.0.0-rc0001 of Cake has been released

This version is the 110th release of Cake, but the first pre-release of Cake 1.0🎉
In this blog post, we will go through some of the major new features of 1.0, which platforms are supported, and how **you** can help to get Cake 1.0 released.

### Why Release Candidate?

> We've chosen to release 1.0 as a pre-release to NuGet and Chocolatey, so we and the community easier can test and ensure that our 1.0 release will be as stable as possible.

Reviewer: ```suggestion
We've chosen to release 1.0 as a pre-release to NuGet and Chocolatey, so we and the community can more easily test and ensure that our 1.0 release will be as stable as possible.
```

> We've chosen to release 1.0 as a pre-release to NuGet and Chocolatey, so we and the community easier can test and ensure that our 1.0 release will be is as stable as possible.

Reviewer 2: `will be as stable as possible`

Author: Addressed

> It will also enable us to see if there are any steps you need to take to adapt your scripts for 1.0, and for us to see if there're any unintended breaking changes, test as many addins, modules, and recipes as possible.

Reviewer: ```suggestion
It will also enable us to see if there are any steps you need to take to adapt your scripts for 1.0, and for us to see if there're any unintended breaking changes, by testing as many addins, modules, and recipes as possible.
```

### Major new features

* .NET 5 support
* C# 9 support
* Implicit bootstrapping
* Rewritten CLI
* Frosting incorporated into main repo
* Improved globbing support
* Improved documentation

> * Bugfixes, increased test coverage, and integration tests.

Reviewer 2: Either `.` at the end of every point or for none

Author: fixed

<!--excerpt-->

### .NET 5 Support

> Cake 1.0 will fully support running natively under .NET 5 both with scripts using the .NET Tool [Cake.Tool](https://www.nuget.org/packages/Cake.Tool/) and as regular .NET Console applications using the [Cake.Frosting](https://www.nuget.org/packages/Cake.Frosting/) library.

Reviewer 2: I would link to https://cakebuild.net/docs/running-builds/runners/dotnet-core-tool and https://cakebuild.net/docs/running-builds/runners/cake-frosting instead of the NuGet page

Author: adjusted

### C# 9 support

With .NET 5 and the latest version of Roslyn comes the full power of C# 9 to your build process.

> And if we may say so, records are brilliant for your build state

Reviewer 2: Trailing `.` missing

Author: added

    string Name,
    string Container,
    string Key
);

public record BuildData(
    string Version,
    DotNetCoreMSBuildSettings MSBuildSettings,
    StorageAccount StorageAccount
)
{
    public bool ShouldPublish { get; } = !string.IsNullOrEmpty(StorageAccount.Name)
                                            && !string.IsNullOrEmpty(StorageAccount.Container)
                                            && !string.IsNullOrEmpty(StorageAccount.Key);
}

Setup<BuildData>(context=>{
    var version = context.Argument("version", "1.0.0");

    return new (
        version,
        new DotNetCoreMSBuildSettings()
                                .WithProperty("Version", version)
                                .WithProperty("Configuration", context.Argument("configuration", "Release")),
        new (
            context.EnvironmentVariable("PUBLISH_STORAGE_ACCOUNT"),
            context.EnvironmentVariable("PUBLISH_STORAGE_CONTAINER"),
            context.EnvironmentVariable("PUBLISH_STORAGE_KEY")
        )
    );
});

Task("Build")
    .Does<BuildData>((context, data) => context.Information("Building {0}", data.Version));

RunTarget("Build");
```

### Implicit bootstrapping

> Boostrapping of Cake modules are now implicit, which means that you no longer will need to execute Cake twice to download and initialize modules, just add the module directive to your script and this will be handled transparently for you.

Reviewer 2: `is now implicit`

Author: adjuested

> Boostrapping of Cake modules are now implicit, which means that you no longer will need to execute Cake twice to download and initialize modules, just add the module directive to your script and this will be handled transparently for you.

Reviewer 2: `to additionally download and initialize modules` (otherwise it sounds like you need to run Cake twice just for downloading in initializing modules)

```c#
#module nuget:?package=Cake.DotNetTool.Module&version=0.4.0
```

### Rewritten CLI

> We're now utilizing [Spectre.Cli](https://www.nuget.org/packages/Spectre.Cli/) for a more robust, flexible and future prof handling of

Reviewer 2: Link to https://spectresystems.github.io/spectre.cli/ instead of NuGet page

> We're now utilizing [Spectre.Cli](https://www.nuget.org/packages/Spectre.Cli/) for a more robust, flexible and future prof handling of

Reviewer 2: End of sentence seems to be missing

Author: 3AM no idea what I was thinking... but added something :)

> We're now utilizing [Spectre.Cli](https://spectresystems.github.io/spectre.cli/) for a more robust, flexible and future prof handling of command line input.

Reviewer: ```suggestion
We're now utilizing [Spectre.Cli](https://spectresystems.github.io/spectre.cli/) for a more robust, flexible and future proof handling of command line input.
```

Full details of everything that was included in this release can be seen below.

> ### Frosting incorporated into main rep

Reviewer: ```suggestion
### Frosting incorporated into main repository
```

> Cake Frosting is now incorporated in same GitHub repository as ( [cake-build/cake](https://github.com/cake-build/cake/) ) all Cake other Cake runners, this means it'll be released and supported with the same cadence as them.

Reviewer: ```suggestion
Cake Frosting is now incorporated in the same GitHub repository as ( [cake-build/cake](https://github.com/cake-build/cake/) ) all the other Cake runners, this means it'll be released and supported with the same cadence as them.
```

### Supported platforms

| Runner | .NET 5 | .NET 4.6.1 | .NET Core 3.1 | .NET Core 3.0 | .NET Core 2.1 | .NET Core 2.0 | .NET Standard 2.0 | Mono |
|--------|--------|------------|---------------|---------------|---------------|---------------|-------------------|------|
| [Cake.Tool](https://www.nuget.org/packages/Cake.Tool/1.0.0-rc0001)           | ✔  | ❌ | ✔  | ✔  | ✔  | ❌ | ❌ | ❌ |
| [Cake.Frosting](https://www.nuget.org/packages/Cake.Frosting/1.0.0-rc0001)   | ✔  | ✔  | ✔  | ✔  | ✔  | ✔  | ✔  | ✔  |
| [Cake](https://www.nuget.org/packages/Cake/1.0.0-rc0001)                     | ❌ | ✔  | ❌ | ❌ | ❌ | ❌ | ❌ | ✔  |

> | [Cake.CoreCLR](https://www.nuget.org/packages/Cake.CoreCLR/1.0.0-rc0001)     | ❌ | ❌ | ❌ | ❌ | ❌  | ✔ | ❌ | ❌ |

Reviewer 2: Use the same naming for runners as on [documentation](https://cakebuild.net/docs/running-builds/runners/) and link to the runner documentation instead of NuGet package.

Author: adjusted

### Supported operating systems

| Runner | macOS | Linux | Windows |
| ------ | ----- | ----- | ------- |
| [Cake.Tool](https://www.nuget.org/packages/Cake.Tool/1.0.0-rc0001)           | ✔  | ✔  | ✔  |
| [Cake.Frosting](https://www.nuget.org/packages/Cake.Frosting/1.0.0-rc0001)   | ✔  | ✔  | ✔  |
| [Cake](https://www.nuget.org/packages/Cake/1.0.0-rc0001)                     | ✔  | ✔  | ✔  |

> | [Cake.CoreCLR](https://www.nuget.org/packages/Cake.CoreCLR/1.0.0-rc0001)     | ✔  | ✔  | ✔  |

Reviewer 2: Use the same naming for runners as on [documentation](https://cakebuild.net/docs/running-builds/runners/) and link to the runner documentation instead of NuGet package.

Author: adjuested

### Supported CI Systems

| Runner | AppVeyor | Azure Pipelines | Bamboo | Bitbucket Pipelines | Bitrise | Continua CI | GitHub Actions | GitLab CI | GoCD | Jenkins | MyGet | TeamCity | TravisCI |
| - | - | - | - | - | - | - | - | - | - | - | - | - | - |
| [Cake.Tool](https://www.nuget.org/packages/Cake.Tool/1.0.0-rc0001)           | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
| [Cake.Frosting](https://www.nuget.org/packages/Cake.Frosting/1.0.0-rc0001)   | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
| [Cake](https://www.nuget.org/packages/Cake/1.0.0-rc0001)                     | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |

> | [Cake.CoreCLR](https://www.nuget.org/packages/Cake.CoreCLR/1.0.0-rc0001)     | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |

Reviewer 2: Use the same naming for runners as on [documentation](https://cakebuild.net/docs/running-builds/runners/) and link to the runner documentation instead of NuGet package.

Author: adjusted

### How can you help?

Help us by taking Cake 1.0.0 RC 1 for spin, test your scripts, addins and modules, if you have have questions please raise them on our [GitHub Discussions](https://github.com/cake-build/cake/discussions) and report any issues on our [GitHub Issues](https://github.com/cake-build/cake/issues).

> Easiest and safest way to test is by using [Cake.Tool](https://www.nuget.org/packages/Cake.Tool/1.0.0-rc0001) with a local tool manifest.

Reviewer 2: Link to https://cakebuild.net/docs/running-builds/runners/dotnet-core-tool instead of nuget.org page

Author: adjusted

> Easiest and safest way to test is by using [.NET Core Tool](https://cakebuild.net/docs/running-builds/runners/dotnet-core-tool) with a local tool manifest.

Reviewer: ```suggestion
Easiest and safest way to test is by using [.NET Tool](https://cakebuild.net/docs/running-builds/runners/dotnet-core-tool) with a local tool manifest.
```
